cups/ppd-cache.c

```diff
@@ -3790,6 +3790,9 @@ _ppdCreateFromIPP(char   *buffer,	/* I - Filename buffer */
 	PRINTF_COLOROPTION("RGB", _("Color"), CUPS_CSPACE_SRGB, 8)
 
 	default_color = "RGB";
+
+	if (ippGetCount(attr) == 1 || !ippContainsString(attr, "sgray_8") && !ippContainsString(attr, "black_1") && !ippContainsString(attr, "black_8"))
```

Comment:
I think the count of 1 check is unnecessary given you are checking all of the other grayscale color spaces. But if we leave it in place I'd want to add parens around the ippContainsString checks (yes, it's fine for C order of operations but parens make it explicit visually...)

Reply:
@michaelrsweet My intention with the count was the shortcut execution - meaning the code will not need to check for all strings if there is only 1 keyword - so ```ippGetCount()``` will return ```1``` and ```ippContainsString()``` calls will be skipped.

Did I misunderstand C condition's evaluation or do you think it is not worth it there?

Reply:
@zdohnal If the ippContainsString calls were expensive or this code was executed many times, then doing the initial shortcut evaluation would be worthwhile. But in this case it is a one-time set of calls so the extra optimization is unnecessary.

Reply:
Ok, got it! Thanks for the merging!